xknx/cemi/cemi_frame.py

```diff
@@ -195,58 +220,459 @@ def from_knx_data_link_layer(cemi: bytes, code: CEMIMessageCode) -> CEMIFrame:
-            tpdu[0] |= self.tpci.to_knx()
-            npdu_len = self.payload.calculated_length()
+        """Serialize to CEMI raw data."""
+        return (
+            self.object_type.value.to_bytes(2, "big")
+            + self.object_instance.to_bytes(1, "big")
+            + self.property_id.to_bytes(1, "big")
+            + ((self.number_of_elements << 12) + self.start_index).to_bytes(2, "big")
+        )
 
+    def _property_repr(self) -> str:
         return (
-            bytes(
-                (
-                    self.code.value,
-                    0x00,  # Additional information length
-                )
+            f'object_type="{self.object_type.value}" '
+            f'object_instance="{self.object_instance}" '
+            f'property_id="{self.property_id}" '
+            f'number_of_elements="{self.number_of_elements}" '
+            f'start_index="{self.start_index}" '
+        )
+
+    def __eq__(self, other: object) -> bool:
+        """Equal operator."""
+        return self.__dict__ == other.__dict__
+
+
+class CEMIMPropReadRequest(CEMIData):
+    """Representation of CEMI Device Management Property Read Request."""
+
+    def __init__(self, *, property_info: CEMIMPropInfo):
+        """Initialize CEMIMPropReadRequest object."""
+        self.property_info = property_info
+
+    def calculated_length(self) -> int:
+        """Get length of CEMI data."""
+        return self.property_info.calculated_length()
+
+    def to_knx(self) -> bytes:
+        """Serialize to CEMI raw data."""
+        return self.property_info.to_knx()
+
+    @classmethod
+    def from_knx(cls, raw: bytes) -> CEMIData:
+        """Parse/deserialize from KNX/IP raw data."""
+        return cls(property_info=CEMIMPropInfo.from_knx(raw))
+
+    def __repr__(self) -> str:
+        """Return object as readable string."""
+        return f"CEMIMPropReadRequest({self.property_info._property_repr()})"
+
+
+class CEMIMPropReadResponse(CEMIData):
+    """Representation of CEMI Device Management Property Read Response."""
+
+    def __init__(
+        self,
+        *,
+        property_info: CEMIMPropInfo,
+        data: bytes,
+    ):
+        """Initialize CEMIMPropReadResponse object."""
+        self.property_info = property_info
+        self.data = data
+
+    @property
+    def error_code(self) -> CEMIErrorCode | None:
+        """Return an optional CEMI error code."""
+        if self.property_info.number_of_elements == 0:
+            return CEMIErrorCode(self.data[0])
+
+        return None
+
+    def calculated_length(self) -> int:
+        """Get length of CEMI data."""
+        return CEMIMPropInfo.LENGTH + len(self.data)
+
+    def to_knx(self) -> bytes:
+        """Serialize to CEMI raw data."""
+        return self.property_info.to_knx() + self.data
+
+    @classmethod
+    def from_knx(cls, raw: bytes) -> CEMIData:
+        """Parse/deserialize from KNX/IP raw data."""
+        if len(raw) < 7:
```

Comment:
```suggestion
        if len(raw) <= CEMIMPropInfo.LENGTH:
```
We should use constants for readability when possible.
You could also define a MIN_LENGTH class attribute if you prefer.